src/ethereum_test_tools/exceptions/evmone_exceptions.py

```diff
@@ -1,51 +1,74 @@
-        EOFException.INVALID_MAGIC: "err: invalid_prefix",
-        EOFException.INVALID_FIRST_SECTION_TYPE: "err: invalid_first_section_type",
-        EOFException.INVALID_SECTION_BODIES_SIZE: "err: invalid_section_bodies_size",
-        EOFException.INVALID_TYPE_SIZE: "err: invalid_type_section_size",
-        EOFException.INCOMPLETE_SECTION_SIZE: "err: incomplete_section_size",
-        EOFException.INCOMPLETE_SECTION_NUMBER: "err: incomplete_section_number",
-        EOFException.TOO_MANY_CODE_SECTIONS: "err: too_many_code_sections",
-        EOFException.ZERO_SECTION_SIZE: "err: zero_section_size",
-    }
-
-    # Reverse the message_map to create a new dictionary where values become keys
-    reverse_message_map = {v: k for k, v in message_map.items()}
-
-    def __init__(self):
-        pass
-
-    def parse_exception(self, exception: EOFException) -> str:
+    _mapping_data = (
+        # TODO EVMONE needs to differentiate when the section is missing in the header or body
+        ExceptionMessage(EOFException.MISSING_STOP_OPCODE, "err: no_terminating_instruction"),
+        ExceptionMessage(EOFException.MISSING_CODE_HEADER, "err: code_section_missing"),
+        ExceptionMessage(EOFException.MISSING_TYPE_HEADER, "err: type_section_missing"),
+        # TODO EVMONE these exceptions are too similar, this leeds to ambiguity
+        ExceptionMessage(EOFException.MISSING_TERMINATOR, "err: header_terminator_missing"),
+        ExceptionMessage(
+            EOFException.MISSING_HEADERS_TERMINATOR, "err: section_headers_not_terminated"
+        ),
+        ExceptionMessage(EOFException.INVALID_VERSION, "err: eof_version_unknown"),
+        ExceptionMessage(EOFException.INVALID_MAGIC, "err: invalid_prefix"),
+        ExceptionMessage(
+            EOFException.INVALID_FIRST_SECTION_TYPE, "err: invalid_first_section_type"
+        ),
+        ExceptionMessage(
+            EOFException.INVALID_SECTION_BODIES_SIZE, "err: invalid_section_bodies_size"
+        ),
+        ExceptionMessage(EOFException.INVALID_TYPE_SIZE, "err: invalid_type_section_size"),
+        ExceptionMessage(EOFException.INCOMPLETE_SECTION_SIZE, "err: incomplete_section_size"),
+        ExceptionMessage(EOFException.INCOMPLETE_SECTION_NUMBER, "err: incomplete_section_number"),
+        ExceptionMessage(EOFException.TOO_MANY_CODE_SECTIONS, "err: too_many_code_sections"),
+        ExceptionMessage(EOFException.ZERO_SECTION_SIZE, "err: zero_section_size"),
+    )
+
+    def __init__(self) -> None:
+        assert len(set(entry.exception for entry in self._mapping_data)) == len(
+            self._mapping_data
+        ), "Duplicate exception in _mapping_data"
+        assert len(set(entry.message for entry in self._mapping_data)) == len(
+            self._mapping_data
+        ), "Duplicate message in _mapping_data"
+        self.exception_to_message_map: frozenbidict = frozenbidict(
+            {entry.exception: entry.message for entry in self._mapping_data}
```

Comment:
So this cycle will iterate over all known exceptions for each test we generate cause we init a new i stance of mapper after filling the fixture?

Reply:
Good point, I wanted to avoid defining it directly in the class definition as it's difficult to debug the asserts (as they hit upon import). 

Perhaps we could use your singleton trick here?